### ISSUE TITLE
Add support for overriding the redirectUrl in `loginWithRedirect`

### ISSUE THREAD

**Issue by jangolano** [salte-auth/salte-auth#229]
### Description
When using a provider that has a separate loginurl and log out url, the redirect back to the login url isn't occurring, instead they are being redirected to the location.href.

### Expected outcome
The user should be redirected to the login redirect url.

### Actual outcome
The user is being redirected to the location.href instead of the login redirect url.

### Browsers Affected

- [X] Chrome
- [X] Firefox
- [X] Safari 10
- [X] Safari 9
- [X] Safari 8
- [X] Safari 7
- [X] Edge
- [X] IE 11
- [X] IE 10

**Comment by cecilia-sanare** [comment]
This is intended behavior.

Basically this ensures the user is brought back to where they started when using `loginWithRedirect`.

https://github.com/salte-io/salte-auth/blob/master/src/salte-auth.js#L124-L132

### Example

* User hits https://salte.io/my-custom-view
* User is redirected to the IDP and logs in
* User is redirected back to https://salte.io
* `salte-auth` then sends the user back to their initial request, 
which in this case was https://salte.io/my-custom-view

**Comment by jangolano** [comment]
This doesn't work well in our case.  
The user https://myhomepage
The user is redirected to the IDP and logs in
The user is redirected to https://myhompage/dashboard (which is the redirect url sent to the IDP)
Then another redirect occurs send the user back to https://myhomepage   The user needs to land on https://myhomepage/dashboard after logging in.
@cecilia-sanare   What about taking in a list of urls that the alternative behavior would be invoked for?

**Comment by cecilia-sanare** [comment]
One option would be to allow the developer to provide an override `redirectUrl` to `loginWithRedirect`.

```js
const auth = new SalteAuth({
  // ...
});

auth.loginWithRedirect('https://myhomepage/dashboard'); // This will take the user to /dashboard
auth.loginWithRedirect(); // This will take the user back where they started
```

**Comment by jangolano** [comment]
@cecilia-sanare 
I am good with this approach if you are?  I can open a PR for your review.

**Comment by cecilia-sanare** [comment]
This is going to take a few more minutes to deploy.

It looks like NPM had to invalidate everyones tokens due to a [security incident](https://status.npmjs.org/incidents/dn7c1fgrr7ng).
So now I have to regenerate the token for @salte-bot.

**Comment by cecilia-sanare** [comment]
This change is now available in [v2.10.0](https://github.com/salte-io/salte-auth/releases/tag/v2.10.0)!
